src/ILLink.RoslynAnalyzer/DataFlow/LocalDataFlowVisitor.cs

```diff
@@ -89,60 +97,100 @@ public override TValue VisitLocalReference (ILocalReferenceOperation operation,
-				// Doesn't get called for assignments to locals, which are handled above.
-				HandleAssignment (value, targetValue, operation);
-				break;
-			case IPropertyReferenceOperation propertyRef:
-				// A property assignment is really a call to the property setter.
-				var setMethod = propertyRef.Property.SetMethod;
-				if (setMethod == null) {
-					// This can happen in a constructor - there it is possible to assign to a property
-					// without a setter. This turns into an assignment to the compiler-generated backing field.
-					// To match the linker, this should warn about the compiler-generated backing field.
-					// For now, just don't warn.
-					break;
+			case IParameterReferenceOperation: {
+					// These assignment targets have the same dataflow representation for LValues and RValues,
+					// so we can use the target value that we already computed. This simplification is specific
+					// to the trim analysis, so would ideally be moved into TrimAnalysisVisitor.
+					TValue value = Visit (operation.Value, state);
+					HandleAssignment (value, targetValue, operation);
+					return value;
 				}
-				TValue instanceValue = Visit (propertyRef.Instance, state);
-				// The return value of a property set expression is the value,
-				// even though a property setter has no return value.
-				HandleMethodCall (
-					setMethod,
-					instanceValue,
-					ImmutableArray.Create (value),
-					operation);
-				break;
-			// TODO: when setting a property in an attribute, target is an IPropertyReference.
-			case IArrayElementReferenceOperation arrayElementRef:
-				if (arrayElementRef.Indices.Length != 1)
-					break;
 
-				HandleArrayElementWrite (Visit (arrayElementRef.ArrayReference, state), Visit (arrayElementRef.Indices[0], state), value, operation);
-				break;
+			// The remaining cases don't have a dataflow value that represents LValues, so we need
+			// to handle the LHS specially.
+			case IPropertyReferenceOperation propertyRef: {
+					var setMethod = propertyRef.Property.SetMethod;
+					if (setMethod == null) {
+						// This can happen in a constructor - there it is possible to assign to a property
+						// without a setter. This turns into an assignment to the compiler-generated backing field.
+						// To match the linker, this should warn about the compiler-generated backing field.
+						// For now, just don't warn. https://github.com/dotnet/linker/issues/2731
+						break;
+					}
+					TValue instanceValue = Visit (propertyRef.Instance, state);
+					TValue value = Visit (operation.Value, state);
+					HandleMethodCall (
+						setMethod,
+						instanceValue,
+						ImmutableArray.Create (value),
+						operation);
+					// The return value of a property set expression is the value,
+					// even though a property setter has no return value.
+					return value;
+				}
+			// TODO: when setting a property in an attribute, target is an IPropertyReference.
```

Comment:
Do we have an issue tracking this?

Reply:
Good point, thanks: https://github.com/dotnet/linker/issues/2751